### ISSUE TITLE
openssl-req: unexpected warning and errors when reading CSRs

### ISSUE THREAD

**Issue by Fantaztig** [openssl/openssl#25125]
We found that after upgrading to openssl 3 we get some unexpected errors in our environments when working with openssl req through PHP bindings.

Equivalent CLI call before (openssl 1.x):
```
cat csr.pem | openssl req -subject
subject=C = US, ST = Test, L = Test, O = Test, OU = Test, CN = Test, emailAddress = Test@Test.com
-----BEGIN CERTIFICATE REQUEST-----
MIICrDCCAZQC...rPcoHF950g==
-----END CERTIFICATE REQUEST-----
```
Equivalent CLI call after (openssl 3.3.1):
```
cat csr.pem | openssl req -subject
Warning: Will read cert request from stdin since no -in option is given
Can't open "/xxx/openssl/openssl.cnf" for reading, No such file or directory
40D7A695E57F0000:error:80000002:system library:BIO_new_file:No such file or directory:crypto/bio/bss_file.c:67:calling fopen(/xxx/openssl/openssl.cnf, r)
40D7A695E57F0000:error:10000080:BIO routines:BIO_new_file:no such file:crypto/bio/bss_file.c:75:
```
Not having the openssl.cnf in the default location is a constraint of our environment and it has worked like that for years before.
We supply the config parameter for all operations where it is possible and makes sense. However, for many operations like openssl_csr_get_subject it's not possible to supply the config as an argument in PHP, nor is it really expected.

Imho the operation at least shouldn't fail in this case and analogous to #22893 openssl req shouldn't print a warning when piping/reading a CSR from redirected stdin.
I'm not sure about whether requiring the openssl.cnf for every operation has some deeper reason behind it or if it could also be optional for commands that don't really need it.

**Comment by mattcaswell** [comment]
Agreed this looks very similar to #22893. I assume a fix along the lines of #23526 would resolve it.

Adding "help wanted" label.

**Comment by zoeychappell** [comment]
Hello, 

I'm interested in taking a look at this!

@bbbrumley

**Comment by zoeychappell** [comment]
It looks like this issue was resolved by @bhaskar1001101 in PR #25179. Since that has already been merged into the main branch, this issue can be closed!
**Fix**: 

```
 else if (!newreq && isatty(fileno_stdin()))
            BIO_puts(bio_err,
                "Warning: Will read cert request from stdin since no -in option is given\n");
    }

```
I verified the fix locally on Windows 11 (WSL2) 

**Test results:**

1. Piped input (No warning): `cat csr.pem | LD_LIBRARY_PATH=. OPENSSL_CONF=./apps/openssl.cnf ./apps/openssl req -subject` Output: `subject=C=US, ST=Test, L=Test, O=Test, CN=Test`

2. Interactive input (Warning shown): `LD_LIBRARY_PATH=. OPENSSL_CONF=./apps/openssl.cnf ./apps/openssl req -subject` Output: `Warning: Will read cert request from stdin since no -in option is given`

@bbbrumley

**Comment by mattcaswell** [comment]
Thanks @zoeychappell for confirming. Closing.
